CalibTracker/SiStripHitEfficiency/plugins/HitEff.cc

```diff
@@ -369,15 +377,55 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
 
       // Check whether the trajectory has some missing hits
       bool hasMissingHits = false;
-      for (itm = TMeas.begin(); itm != TMeas.end(); itm++) {
-        auto theHit = (*itm).recHit();
+      unsigned int previous_layer = 999;
+      vector<unsigned int> missedLayers;
+
+      for (const auto& itm : TMeas) {
+        auto theHit = itm.recHit();
+        unsigned int iidd = theHit->geographicalId().rawId();
+        unsigned int layer = ::checkLayer(iidd, tTopo);
+        int missedLayer = int(layer + 1);
+        int diffPreviousLayer = int(layer - previous_layer);
```

Comment:
As such this cannot work: making a difference of two `unsigned int` can never return a negative number, see L391 below.
At the very minimum this could be
```suggestion
        int diffPreviousLayer = int(layer) - int(previous_layer);
```
but probably redefining `layer` and `previous_layer` as `int` since the beginning could avoid some gymnastic

Reply:
Ok, I redefined those as ints so the rest will be changed aswell. Thanks